File: pset1/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int get_index(string);
void print_grade(int);

int main(void)
{
    string text = get_string("Text:");
    int index = get_index(text);
    print_grade(index);
}

int get_index(string text)
{
    const char *ptext = text;
    int word_cnt = 0;
    int letter_cnt = 0;
    int sentence_cnt = 0;

    for (int i = 0; i < strlen(text); i++, ptext++)
    {
        if (isalpha(*ptext))
        {
            letter_cnt += 1;
        }
        else
        {
            if (*ptext == ' ')
            {
                word_cnt += 1;
            }
            else if (*ptext == '.' || *ptext == '?' || *ptext == '!')
            {
                sentence_cnt += 1;

                if (i == strlen(text) - 1)
                {
                    word_cnt += 1;
                }
            }
        }
    }

    float L = (float) letter_cnt / word_cnt * 100;
    float S = (float) sentence_cnt / word_cnt * 100;
    float index = 0.0588 * L - 0.296 * S - 15.8;

    return (round(index));
}

void print_grade(int index)
{
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

File: pset6/readability/readability.py
import cs50


def main():
    text = cs50.get_string("Text:")

    N = 0  # num words
    L = 0  # num letters
    S = 0  # num sentences

    text = text.split()
    N = len(text)
    for word in text:
        for t in word:
            if (t == ".") or (t == "!") or (t == "?"):
                S += 1
            elif str.isalpha(t):
                L += 1

    print("#words:", N, "#sentences:", S, "#letters:", L)

    index = 0.0588 * (L / (N / 100.0)) - 0.296 * (S / (N / 100.0)) - 15.8
    index_rnd = round(index)

    if index < 1:
        print("Before Grade 1")
    elif index_rnd >= 16:
        print("Grade 16+")
    else:
        print("Grade", index_rnd)


if __name__ == "__main__":
    main()